src/main/java/uk/gov/hmcts/reform/civil/handler/callback/user/RespondToClaimCallbackHandler.java

```diff
@@ -159,31 +191,99 @@ private CallbackResponse setApplicantResponseDeadline(CallbackParams callbackPar
             .build();
     }
 
+    //TODO: 2nd DQ for 2nd solicitor
+    //TODO: how best to store DQ when one solicitor representing both
     private CallbackResponse setApplicantResponseDeadlineV1(CallbackParams callbackParams) {
         CaseData caseData = callbackParams.getCaseData();
-        LocalDateTime responseDate = time.now();
-        AllocatedTrack allocatedTrack = caseData.getAllocatedTrack();
+
+        // persist respondent address (ccd issue)
+        var updatedRespondent1 = caseData.getRespondent1().toBuilder()
+            .primaryAddress(caseData.getRespondent1Copy().getPrimaryAddress())
+            .build();
 
         CaseData.CaseDataBuilder updatedData = caseData.toBuilder()
-            .respondent1ResponseDate(responseDate)
-            .applicant1ResponseDeadline(getApplicant1ResponseDeadline(responseDate, allocatedTrack))
-            .businessProcess(BusinessProcess.ready(DEFENDANT_RESPONSE));
+            .respondent1(updatedRespondent1)
+            .respondent1Copy(null);
 
-        // moving statement of truth value to correct field, this was not possible in mid event.
-        StatementOfTruth statementOfTruth = caseData.getUiStatementOfTruth();
-        Respondent1DQ dq = caseData.getRespondent1DQ().toBuilder()
-            .respondent1DQStatementOfTruth(statementOfTruth)
-            .build();
+        // if present, persist the 2nd respondent address in the same fashion as above, i.e ignore for 1v1
+        if (ofNullable(caseData.getRespondent2()).isPresent()
+            && ofNullable(caseData.getRespondent2Copy()).isPresent()) {
+            var updatedRespondent2 = caseData.getRespondent2().toBuilder()
+                .primaryAddress(caseData.getRespondent2Copy().getPrimaryAddress())
+                .build();
 
-        updatedData.respondent1DQ(dq);
-        // resetting statement of truth to make sure it's empty the next time it appears in the UI.
-        updatedData.uiStatementOfTruth(StatementOfTruth.builder().build());
+            updatedData.respondent2(updatedRespondent2).respondent2Copy(null);
+        }
+
+        LocalDateTime responseDate = time.now();
+        AllocatedTrack allocatedTrack = caseData.getAllocatedTrack();
+
+        // same legal rep - will respond for both and set applicant 1 response deadline
+        if (respondent2HasSameLegalRep(caseData)) {
```

Comment:
Ideally we would replace this with a switch statement replying solely on flow flags but as we agreed this would require a significant refactoring of the state flow engine we will have to accept the logic sitting here for the time being.